### PR TITLE
Refresh universes before trying to coerce.

### LINKED ISSUE #6682: Universe Polymorphism and Canonical Structures containing `Type`
#### Version
8.8+alpha (and 8.7.1)

#### Description of the problem
The following snippet fails. I expected it to succeed (after https://github.com/coq/coq/issues/5368 was fixed, at least).

```Coq
Set Universe Polymorphism. Unset Universe Minimization ToSet.
Structure NoProof := no_proof { np_type : Type }.
Canonical Structure NP_Type := no_proof Type. 
Check (fun np (T : np_type np) => I) _ Type.
```

### PR DESCRIPTION
**Pull request by SkySkimmer**
Fix #6682.
Bench waiting jenkins 595.

### CODE REVIEW

File: pretyping/coercion.ml [hunk 1]
```diff
@@ -521,6 +521,13 @@ let rec inh_conv_coerce_to_fail ?loc env evd rigidonly v t c1 =
 
 (* Look for cj' obtained from cj by inserting coercions, s.t. cj'.typ = t *)
 let inh_conv_coerce_to_gen ?loc resolve_tc rigidonly env evd cj t =
+  let evd, cj =
+    let evd, uj_type = Evarsolve.refresh_universes
+        ~status:Evd.univ_flexible ~onlyalg:true (Some false)
+        env evd cj.uj_type
+    in
+    evd, {cj with uj_type }
+  in
```
Comment (SkySkimmer) [inline pretyping/coercion.ml]: I don't know if this is the right place to put this, we could also go for `Pretyping.inh_conv_coerce_to_tycon` (with possibility of missing other callers).
Here seems OK though.

**Comment by SkySkimmer** [comment]
Breaks developments, will investigate.

**Comment by SkySkimmer** [comment]
I can't figure out where the incompatibility comes from exactly, don't know how to progress.
@mattam82 WDYT?

File: pretyping/coercion.ml (hunk 1 above)
Comment (mattam82) [reply pretyping/coercion.ml]: It sounds quite strong to always refresh, while leq can handle cj.uj_type being an algebraic.

**Pull request closed without merging**